docs/src/grids.md

```diff
@@ -0,0 +1,646 @@
+# Grids and architectures
+
+```@meta
+DocTestSetup = quote
+    using Oceananigans
+end
+```
+
+Oceananigans simulates the dynamics of ocean-flavored fluids by solving differential equations that conserve momentum, mass, and energy on a mesh of finite volumes or "cells".
+The first decision we make when setting up a simulation is: on what _grid_ are we going to run our simulation?
+The "grid" encodes fundamental information: the shape of the domain we're simulating in, the way that domain is divided into a mesh of finite volumes, the machine architecture (CPU, GPU, lots of GPUs), and the precision of the numbers we would like to use to represent ocean-flavored fluid variables (double precision or single precision).
+
+One of the simplest grids we can make divides a three-dimensional rectangular domain -- or "a box" --- into evenly-spaced cells.
+To create such a grid on the CPU (the machine that we typically run julia on, basically), we write
+
+```jldoctest grids
+architecture = CPU()
+
+grid = RectilinearGrid(architecture,
+                       topology = (Periodic, Periodic, Bounded),
+                       size = (16, 8, 4),
+                       x = (0, 64),
+                       y = (0, 32),
+                       z = (0, 8))
+
+# output
+16×8×4 RectilinearGrid{Float64, Periodic, Periodic, Bounded} on CPU with 3×3×3 halo
+├── Periodic x ∈ [0.0, 64.0) regularly spaced with Δx=4.0
+├── Periodic y ∈ [0.0, 32.0) regularly spaced with Δy=4.0
+└── Bounded  z ∈ [0.0, 8.0]  regularly spaced with Δz=2.0
+```
+
+This simple grid
```

Comment:
```suggestion
This simple grid:
```

Reply:
"this simple grid: uses the CPU" isn't correct right?

Reply:
see here:

https://style.mla.org/vertical-lists-in-mla-style/

under "Lists That Continue the Sentence Introducing Them"

Reply:
great! even if it was merely your preference I'd be convinced!